clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -100,13 +100,13 @@ public FetchResponse(Map<TopicPartition, PartitionData> responseData, int thrott
         super(new Struct(CURRENT_SCHEMA));
         initCommonFields(responseData);
         struct.set(THROTTLE_TIME_KEY_NAME, throttleTime);
-        this.responseData = responseData;
+        this.responseData = new LinkedHashMap<TopicPartition, PartitionData>(responseData);
```

Comment:
It would be clearer if we can have a constructor for v3 using LinkedHashMap. This may not be possible though. So, it would be useful to at least add a comment that for v3, we expect the caller to pass in a LinkedHashMap since the ordering is relevant.